src/main/java/seedu/address/logic/commands/ChangePasswordCommand.java

```
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Forward hashes the user input password and checks if it matches with the encrypted password saved
+     */
+    private boolean isOldPasswordCorrect() {
+        String forwardHashedInputPassword = forwardHash(oldPassword);
+        String forwardHashActualPassword;
+        UserPrefs userPrefs = model.getUserPrefs();
+        forwardHashActualPassword = userPrefs.getAddressBookEncryptedPassword();
+        if (forwardHashActualPassword.equals(forwardHashedInputPassword)) {
+            logger.info("Actual password and input password matches");
+            return true;
+        } else {
+            logger.warning("Actual password and input password do not match");
+            return false;
+        }
+    }
+
+    /**
+     * Takes new input passwords and checks them against one another.
+     */
+    private boolean isNewPasswordInputsSame() {
+        return newPassword.equals(confirmationPassword);
+    }
+
+    /**
+     * Checks if old password and new password are the same
+     */
+    private boolean isOldAndNewPasswordTheSame() {
+        return oldPassword.equals(newPassword);
+    }
+
+    @Override
+    public CommandResult execute() {
+        // Case where old password is incorrect
+        if (!isOldPasswordCorrect()) {
+            logger.warning("Password is incorrect. Note: Default password is 'password' ");
+            return new CommandResult(MESSAGE_PASSWORD_INCORRECT);
+        }
+
+        // Case where new password and confirmation password do not match
+        if (!isNewPasswordInputsSame()) {
+            logger.warning("New password and confirmation password do not match");
+            return new CommandResult(MESSAGE_PASSWORD_CONFIRMATION_INCORRECT);
+        }
+
+        // Case where old and new passwords are the same
+        if (isOldAndNewPasswordTheSame()) {
+            logger.warning("Old password and new password cannot be the same");
+            return new CommandResult(MESSAGE_OLD_NEW_PS_SAME);
```

Rather than returning a CommandResult, throwing a CommandException might be more appropriate instead as the command did not pass. The logging will not be needed in this case.

---

That is true. I was wondering why it did not highlight red. Did not know about the CommandException. Thanks :)